# Exploring Ebay Car Sales Data

In this guided project, we'll work with a dataset of used cars from eBay Kleinanzeigen, a [classifieds](https://en.wikipedia.org/wiki/Classified_advertising) section of the German eBay website.

The dataset was originally scraped and uploaded to [Kaggle](https://www.kaggle.com/orgesleka/used-cars-database/data). We've made a few modifications from the original dataset that was uploaded to Kaggle:
* We sampled 50,000 data points from the full dataset, to ensure your code runs quickly in our hosted environment
* We dirtied the dataset a bit to more closely resemble what you would expect from a scraped dataset (the version uploaded to Kaggle was cleaned to be easier to work with)

The aim of this project is to clean the data and analyze the included used car listings.


In [27]:
import pandas as pd
import numpy as np

autos = pd.read_csv('autos.csv',encoding="Latin-1")

In [28]:
autos

dateCrawled                                               name  \
0      2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1      2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2      2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3      2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4      2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5      2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6      2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7      2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8      2016-03-22 16:51:34                                         Seat_Arosa   
9      2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   
10     2016-03-15 01:41:36                       VW_Golf_Tuning_in_siber/grau   
11     2016-03-16 18:45:34                         Mercedes_A140_Motorschaden   
12     2016-03-31 19:48:22  Smart_smart_fortwo_coupe_softouch_pure_MHD_Pan...   
13     2016-03-23 10:48:32                                 Audi_A3_1.6_tuning   
14     2016-03-23 11:50:46  Renault_Clio_3__Dynamique_1.2__16_V;_viele_Ver...   
15     2016-04-01 12:06:20        Corvette_C3_Coupe_T_Top_Crossfire_Injection   
16     2016-03-16 14:59:02                                Opel_Vectra_B_Kombi   
17     2016-03-29 11:46:22                          Volkswagen_Scirocco_2_G60   
18     2016-03-26 19:57:44               Verkaufen_mein_bmw_e36_320_i_touring   
19     2016-03-17 13:36:21        mazda_tribute_2.0_mit_gas_und_tuev_neu_2018   
20     2016-03-05 19:57:31  Audi_A4_Avant_1.9_TDI_*6_Gang*AHK*Klimatronik*...   
21     2016-03-06 19:07:10                      Porsche_911_Carrera_4S_Cabrio   
22     2016-03-28 20:50:54                               MINI_Cooper_S_Cabrio   
23     2016-03-10 19:55:34   Peugeot_Boxer_2_2_HDi_120_Ps_9_Sitzer_inkl_Klima   
24     2016-04-03 11:57:02                         BMW_535i_xDrive_Sport_Aut.   
25     2016-03-21 21:56:18  Ford_escort_kombi_an_bastler_mit_ghia_ausstattung   
26     2016-04-03 22:46:28                                Volkswagen_Polo_Fox   
27     2016-03-27 18:45:01             Hat_einer_Ahnung_mit_Ford_Galaxy_HILFE   
28     2016-03-19 21:56:19                                      MINI_Cooper_D   
29     2016-04-02 12:45:44  Mercedes_Benz_E_320_T_CDI_Avantgarde_DPF7_Sitz...   
...                    ...                                                ...   
49970  2016-03-21 22:47:37  c4_Grand_Picasso_mit_Automatik_Leder_Navi_Temp...   
49971  2016-03-29 14:54:12                                         W.Lupo_1.0   
49972  2016-03-26 22:25:23          Mercedes_Benz_Vito_115_CDI_Extralang_Aut.   
49973  2016-03-27 05:32:39                   Mercedes_Benz_SLK_200_Kompressor   
49974  2016-03-20 10:52:31  Golf_1_Cabrio_Tuev_Neu_viele_Extras_alles_eing...   
49975  2016-03-27 20:51:39          Honda_Jazz_1.3_DSi_i_VTEC_IMA_CVT_Comfort   
49976  2016-03-19 18:56:05  Audi_80_Avant_2.6_E__Vollausstattung!!_Einziga...   
49977  2016-03-31 18:37:18                        Mercedes_Benz_C200_Cdi_W203   
49978  2016-04-04 10:37:14                        Mercedes_Benz_E_200_Classic   
49979  2016-03-20 18:38:40                      Volkswagen_Polo_1.6_TDI_Style   
49980  2016-03-12 10:55:54                            Ford_Escort_Turnier_16V   
49981  2016-03-15 09:38:21             Opel_Astra_Kombi_mit_Anhaengerkupplung   
49982  2016-03-29 18:51:08            Skoda_Fabia_4_Tuerer_Bj:2004__85.000Tkm   
49983  2016-03-06 12:43:04                                      Ford_focus_99   
49984  2016-03-31 22:48:48  Student_sucht_ein__Anfaengerauto___ab_2000_BJ_...   
49985  2016-04-02 16:38:23                          Verkaufe_meinen_vw_vento!   
49986  2016-04-04 20:46:02  Chrysler_300C_3.0_CRD_DPF_Automatik_Voll_Ausst...   
49987  2016-03-22 20:47:27  Audi_A3_Limousine_

In [29]:
autos.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

In [30]:
autos.head()

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   

     gearbox  powerPS   model   odometer  monthOfRegistration fuelType  \
0    manuell      158  andere  150,000km                    3      lpg   
1  automatik      286     7er  150,000km                    6   benzin   
2    manuell      102    golf   70,000km                    7   benzin   
3  automatik       71  fortwo   70,000km                    6   benzin   
4    manuell        0   focus  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50

* The dataset contains 20 columns, most of which are strings.
* Some columns have null values, but none have more than ~20% null values.
* The column names use camelcase instead of Python's preferred snakecase, which means we can't just replace spaces with underscores.

In [31]:
autos["price"].unique()

array(['$5,000', '$8,500', '$8,990', ..., '$385', '$22,200', '$16,995'],
      dtype=object)

In [32]:
autos["price"] = autos["price"].str.replace("$","")
autos["price"] = autos["price"].str.replace(",","")
autos["price"] = autos["price"].astype(int)
autos["price"].unique()

array([ 5000,  8500,  8990, ...,   385, 22200, 16995])

In [33]:
autos["odometer"].unique()

array(['150,000km', '70,000km', '50,000km', '80,000km', '10,000km',
       '30,000km', '125,000km', '90,000km', '20,000km', '60,000km',
       '5,000km', '100,000km', '40,000km'], dtype=object)

In [34]:
autos["odometer"] = autos["odometer"].str.replace("km","")
autos["odometer"] = autos["odometer"].str.replace(",","")
autos["odometer"] = autos["odometer"].astype(int)
autos.rename({"odometer":"odometer_km"},axis=1,inplace=True)
autos["odometer_km"].unique()

array([150000,  70000,  50000,  80000,  10000,  30000, 125000,  90000,
        20000,  60000,   5000, 100000,  40000])

** Exploring Price Column **

In [35]:
autos["price"].unique().shape

(2357,)

In [36]:
autos["price"].describe()

count    5.000000e+04
mean     9.840044e+03
std      4.811044e+05
min      0.000000e+00
25%      1.100000e+03
50%      2.950000e+03
75%      7.200000e+03
max      1.000000e+08
Name: price, dtype: float64

In [37]:
autos["price"].value_counts().head()

0       1421
500      781
1500     734
2500     643
1000     639
Name: price, dtype: int64

In [38]:
autos["price"].value_counts().sort_index(ascending=False).head()

99999999    1
27322222    1
12345678    3
11111111    2
10000000    1
Name: price, dtype: int64

In [39]:
autos["price"].value_counts().sort_index(ascending=True).head()

0    1421
1     156
2       3
3       1
5       2
Name: price, dtype: int64

In [40]:
autos.shape

(50000, 20)

** Removing the outliers, that is eliminating all rows having price above 1cr and below 1k **

In [43]:
autos = autos[autos["price"].between(1000,10000000)]

In [44]:
autos.shape

(38633, 20)

** Exploring Odometer_km column **

In [45]:
autos["odometer_km"].unique().shape

(13,)

In [46]:
autos["odometer_km"].describe()

count     38633.000000
mean     122775.218078
std       40800.129272
min        5000.000000
25%      100000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

In [47]:
autos["odometer_km"].value_counts().head()

150000    23317
125000     4341
100000     1861
90000      1569
80000      1334
Name: odometer_km, dtype: int64

In [48]:
autos["odometer_km"].value_counts().sort_index(ascending=True).head()

5000     508
10000    228
20000    692
30000    748
40000    795
Name: odometer_km, dtype: int64

In [49]:
autos["odometer_km"].value_counts().sort_index(ascending=False).head()

150000    23317
125000     4341
100000     1861
90000      1569
80000      1334
Name: odometer_km, dtype: int64

** Odometer column doesn't seems to have any outliers **

In [51]:
autos.shape

(38633, 20)

In [52]:
autos.describe()

price  yearOfRegistration       powerPS    odometer_km  \
count  3.863300e+04        38633.000000  38633.000000   38633.000000   
mean   7.756859e+03         2005.677012    128.934719  122775.218078   
std    5.681184e+04           86.677787    215.903650   40800.129272   
min    1.000000e+03         1000.000000      0.000000    5000.000000   
25%    2.200000e+03         2001.000000     80.000000  100000.000000   
50%    4.350000e+03         2005.000000    116.000000  150000.000000   
75%    8.950000e+03         2009.000000    160.000000  150000.000000   
max    1.000000e+07         9999.000000  17700.000000  150000.000000   

       monthOfRegistration  nrOfPictures    postalCode  
count         38633.000000       38633.0  38633.000000  
mean              5.998706           0.0  51691.117205  
std               3.591257           0.0  25689.360881  
min               0.000000           0.0   1067.000000  
25%               3.000000           0.0  31224.000000  
50%               6.000000           0.0  50858.000000  
75%               9.000000           0.0  72534.000000  
max              12.000000           0.0  99998.000000

#### Working with dates

Let's now move on to the date columns and understand the date range the data covers.

There are 5 columns that should represent date values. Some of these columns were created by the crawler, some came from the website itself. We can differentiate by referring to the data dictionary

   * date_crawled: added by the crawler
   * last_seen: added by the crawler
   * ad_created: from the website
   * registration_month: from the website
   * registration_year: from the website

Right now, the date_crawled, last_seen, and ad_created columns are all identified as string values by pandas. Because these three columns are represented as strings, we need to convert the data into a numerical representation so we can understand it quantitatively.

Let's first understand how the values in the three string columns are formatted. These columns all represent full timestamp values, like so:


In [54]:
autos.columns

Index(['dateCrawled', 'name', 'seller', 'offerType', 'price', 'abtest',
       'vehicleType', 'yearOfRegistration', 'gearbox', 'powerPS', 'model',
       'odometer_km', 'monthOfRegistration', 'fuelType', 'brand',
       'notRepairedDamage', 'dateCreated', 'nrOfPictures', 'postalCode',
       'lastSeen'],
      dtype='object')

In [55]:
autos[['dateCrawled','dateCreated','lastSeen']][0:5]

dateCrawled          dateCreated             lastSeen
0  2016-03-26 17:47:46  2016-03-26 00:00:00  2016-04-06 06:45:54
1  2016-04-04 13:38:56  2016-04-04 00:00:00  2016-04-06 14:45:08
2  2016-03-26 18:57:24  2016-03-26 00:00:00  2016-04-06 20:15:37
3  2016-03-12 16:58:10  2016-03-12 00:00:00  2016-03-15 03:16:28
4  2016-04-01 14:38:50  2016-04-01 00:00:00  2016-04-01 14:38:50

In [56]:
print(autos['dateCrawled'].str[:10].head(15)) #To select the first 10 characters (the date) in each column

0     2016-03-26
1     2016-04-04
2     2016-03-26
3     2016-03-12
4     2016-04-01
5     2016-03-21
7     2016-03-16
12    2016-03-31
13    2016-03-23
14    2016-03-23
15    2016-04-01
17    2016-03-29
19    2016-03-17
20    2016-03-05
21    2016-03-06
Name: dateCrawled, dtype: object


In [57]:
#date_drawled
print(autos['dateCrawled'].value_counts(normalize=True, dropna=False).sort_index().head()) #Using head here to avoid printing walls of data

print(autos['dateCrawled'].value_counts(normalize=True, dropna=False).sort_index(ascending=False).head())

2016-03-05 14:06:30    0.000026
2016-03-05 14:06:40    0.000026
2016-03-05 14:07:21    0.000026
2016-03-05 14:07:26    0.000026
2016-03-05 14:07:40    0.000026
Name: dateCrawled, dtype: float64
2016-04-07 14:36:56    0.000026
2016-04-07 14:36:55    0.000026
2016-04-07 14:36:44    0.000026
2016-04-07 14:30:26    0.000026
2016-04-07 14:30:09    0.000026
Name: dateCrawled, dtype: float64


The date_crawled section is set between the dates March 05 2016 and April 07 2016.

In [58]:
#ad_created
print(autos['dateCreated'].value_counts(normalize=True, dropna=False).sort_index().head()) #Using head here to avoid printing walls of data

print(autos['dateCreated'].value_counts(normalize=True, dropna=False).sort_index(ascending=False).head())

2015-06-11 00:00:00    0.000026
2015-08-10 00:00:00    0.000026
2015-09-09 00:00:00    0.000026
2015-11-10 00:00:00    0.000026
2015-12-30 00:00:00    0.000026
Name: dateCreated, dtype: float64
2016-04-07 00:00:00    0.001320
2016-04-06 00:00:00    0.003365
2016-04-05 00:00:00    0.011985
2016-04-04 00:00:00    0.037326
2016-04-03 00:00:00    0.039448
Name: dateCreated, dtype: float64




   * The ad_created section is set between the dates June 11 2015 and April 03 2016. Which is essentially a bigger bracket than date_crawled
   * On first glance, April 03,04,05 seem to have had a significant numbers of new ads created in comparison with the other days.


In [59]:
#last_seen. This parameter is when the ad was last seen, it could be presumed that it is the date when the car was sold.
print(autos['lastSeen'].value_counts(normalize=True, dropna=False).sort_index().head()) #Using head here to avoid printing walls of data

print(autos['lastSeen'].value_counts(normalize=True, dropna=False).sort_index(ascending=False).head())

2016-03-05 14:46:02    0.000026
2016-03-05 14:49:34    0.000026
2016-03-05 15:16:11    0.000026
2016-03-05 15:16:47    0.000026
2016-03-05 15:28:10    0.000026
Name: lastSeen, dtype: float64
2016-04-07 14:58:50    0.000078
2016-04-07 14:58:48    0.000078
2016-04-07 14:58:46    0.000026
2016-04-07 14:58:45    0.000026
2016-04-07 14:58:44    0.000078
Name: lastSeen, dtype: float64


* A similar date time frame as date_crawled. This could be the presumed date where the car was sold or the ad was taken down altogether.

Let us look at the registration_year series and analyze where we could make some changes, or if any changes are necessary at all.

In [61]:
autos['yearOfRegistration'].describe()

count    38633.000000
mean      2005.677012
std         86.677787
min       1000.000000
25%       2001.000000
50%       2005.000000
75%       2009.000000
max       9999.000000
Name: yearOfRegistration, dtype: float64

Here ofcourse, we need to exclude the year 1000 and 9999. Moreover, any year prior to 1950 seems unlikely and later than 2017 is impossible. So let us clean the data accordingly using the between function like we used in price

In [62]:
autos = autos[autos['yearOfRegistration'].between(1950,2017)]
autos['yearOfRegistration'].value_counts().sort_values().tail()

2007    2259
2003    2477
2004    2608
2006    2652
2005    2785
Name: yearOfRegistration, dtype: int64



With this we can see that most of the cars being sold on eBay at the moment have been registered between 1999-2005 (around 29%)

Around the time, the most used and sold car in Germany (upto this date too in a way) is the VW Golf. If we have to start with a hypothesis it would be that VW Golf should be among the top listed cars considering the demographics and the years of registration. (could also be VW beetle because it picked up around 1999-2003 but highly unlikely because it did go out of style unlike VW Golf)

Brandwise Distribution of Cars

It woudl defintiely be of value for us to see which brands and which models have been listed the most on eBay.

It could help us identify the brand choices of people and how it would have evolved over years.

Here, we will use aggregation to understand the brand column.


In [63]:
a = autos['brand'].value_counts(normalize=True).sort_values(ascending=False)[0:20] #We only need the top 20 manufacturers for now
autos['brand'].value_counts(normalize=True).sort_values(ascending=False)[0:20]

volkswagen        0.212532
bmw               0.124090
mercedes_benz     0.110899
audi              0.097131
opel              0.089567
ford              0.058708
renault           0.037586
peugeot           0.027666
fiat              0.021253
skoda             0.018793
seat              0.017694
smart             0.016804
toyota            0.014631
mazda             0.014396
citroen           0.014003
nissan            0.013689
mini              0.010836
hyundai           0.010757
sonstige_autos    0.010155
volvo             0.008899
Name: brand, dtype: float64



As assumed, VW is the most sold brand not only for new cars in Germany but also for used cars on eBay. (Only logical as usual inflow and outflow with market circulation is statistically and directly proportional) It is followed by the other German bigwigs like bmw, opel, mercedies and the VW descendent audi.

Let us see which of the brands have the highest average price and average odometer count based on this existing data that we have gathered.


In [64]:
autos_mean = {}
autos_drive = {}

for c in a.index:
    selected_rows = autos[autos['brand'] == c]
    
    autos_mean[c] = selected_rows['price'].mean()
    autos_drive[c] = selected_rows['odometer_km'].mean()
    
print(autos_mean)

{'skoda': 6817.6949860724235, 'volkswagen': 6826.665886699508, 'ford': 5737.911725367811, 'mazda': 5229.483636363636, 'audi': 10234.684182161143, 'smart': 3732.797507788162, 'nissan': 6327.812619502868, 'hyundai': 6164.591240875912, 'volvo': 6087.732352941176, 'seat': 5571.693786982249, 'mercedes_benz': 9214.126504602313, 'sonstige_autos': 53478.42525773196, 'bmw': 9338.4349293398, 'opel': 4199.146697837522, 'citroen': 4589.4523364485985, 'peugeot': 3932.067171239357, 'toyota': 5553.971377459749, 'fiat': 3977.514778325123, 'mini': 10640.437198067633, 'renault': 3554.490947075209}




Observation There is a gap between the top 20 prices

   * volksvagen has mid-range prices which makes it the most popular brand

   * sonstige_autos, mini and audi have the highest prices

   * opel and renault have the lowest prices



In [65]:
print(autos_drive)

{'skoda': 110111.42061281337, 'volkswagen': 126008.62068965517, 'ford': 119843.95898350424, 'mazda': 120490.90909090909, 'audi': 127782.26893020749, 'smart': 98668.22429906542, 'nissan': 110860.4206500956, 'hyundai': 102238.44282238443, 'volvo': 137382.35294117648, 'seat': 116834.31952662722, 'mercedes_benz': 130273.77861694596, 'sonstige_autos': 90489.69072164949, 'bmw': 132113.4781691626, 'opel': 124095.55815312683, 'citroen': 114560.7476635514, 'peugeot': 122455.0614947966, 'toyota': 114168.15742397138, 'fiat': 108282.0197044335, 'mini': 89468.59903381643, 'renault': 121740.94707520891}




Here is where things get interesting:

   * The top 2 cars with the highest price average mini and sonstige (Miscellaneous) have the least mileage readings on the odometer.
   * volvo is the brand with the most mileage on the list with bmw, audi and opel closely behind.



In [66]:
# create a dataframe of the mean price and mean mileage to make it easy to compare

mean_price = pd.Series(autos_mean)
mean_mileage = pd.Series(autos_drive)

# create a dataframe of both series
brand_df = pd.DataFrame({'mean_price': mean_price, 'mean_mileage': mean_mileage})

brand_df

mean_mileage    mean_price
audi            127782.268930  10234.684182
bmw             132113.478169   9338.434929
citroen         114560.747664   4589.452336
fiat            108282.019704   3977.514778
ford            119843.958984   5737.911725
hyundai         102238.442822   6164.591241
mazda           120490.909091   5229.483636
mercedes_benz   130273.778617   9214.126505
mini             89468.599034  10640.437198
nissan          110860.420650   6327.812620
opel            124095.558153   4199.146698
peugeot         122455.061495   3932.067171
renault         121740.947075   3554.490947
seat            116834.319527   5571.693787
skoda           110111.420613   6817.694986
smart            98668.224299   3732.797508
sonstige_autos   90489.690722  53478.425258
toyota          114168.157424   5553.971377
volkswagen      126008.620690   6826.665887
volvo           137382.352941   6087.732353



Mini and sonstige (other) have the least odometer readings and the highest prices. Which could help us levitate towards the possibility of higher prices for used cars which have a lesser odometer reading (Which would imply that they have travalled lesser and have relagively healthier engine)

However, from a very high level, the point which more clearly justifies the price is actually (and more importantly, obviously) the brand.

Opel, Renault and Fiat are the reasonable cars, while BMW, Audi and Mercedes are the costlier ones.

VW on the other hand is placed very well in the middle strata of the market and has the highest sales.

The most commonly sold model

We proposed an assumption towards the beginnning that as golf should be one of the best sellers on eBay too as it is one of the best sellers in Germany overall.

Also, let us take a look into the more interesting brands and what models are most popular for brands like VW, Audi, BMW, Opel and Mercedes Benz

To do this we will look into the column labelled model

Let's start off with the models which are popular overall


In [67]:
autos['model'].value_counts().sort_values(ascending=False).head(6)

golf      3040
andere    2694
3er       2317
passat    1172
a4        1105
5er       1078
Name: model, dtype: int64

### Golf is the most sold model overall!

In [68]:
#The most preferred models brandwise
brands = ['volkswagen','bmw','audi','mercedes_benz','opel']

for a in brands:
    models_car = autos[autos['brand']==a]
    b = models_car['model'].value_counts().sort_values(ascending=False).head(1)
    print('{0} brand : {1}'.format(a,b))
    print('\n')

volkswagen brand : golf    3040
Name: model, dtype: int64


bmw brand : 3er    2317
Name: model, dtype: int64


audi brand : a4    1105
Name: model, dtype: int64


mercedes_benz brand : c_klasse    1051
Name: model, dtype: int64


opel brand : astra    1007
Name: model, dtype: int64




The Most popular models by brand!

   * VW - Golf
   * BMW - 3er
   * Opel - Corsa
   * Audi - A4
   * Mercedes Benz - C Class

In [69]:
autos['notRepairedDamage'].describe()

count     32608
unique        2
top        nein
freq      30241
Name: notRepairedDamage, dtype: object

In [70]:
ohne_repair_mean = autos.loc[autos['notRepairedDamage']=="nein",'price'].mean()
mit_repair_mean = autos.loc[autos['notRepairedDamage']=="ja",'price'].mean()

print(ohne_repair_mean)
print(mit_repair_mean)

8546.795939287722
4272.467680608365


Quite an evident differnce between the price of cars without repair and the ones with repair.

The ones without repair have a clearly higher price (3 times!)



## Conclusion

The inferences from this project are unique as well as sometimes directly aligning to everything with what we ideally consider to be common knowledge.

** Observations **

   * VW is the most sold brand for new as well as used cars with Gold being in the lead.
   * This is followed by Opel, another Deutschland classic!
   * Opel and Renault have the lowest prices and BMW, Audi and Mercedes are amongst the highest.
   * A significant chunk (~30%) of second hand cars which were tried to be sold in the used car market from 2015-2016 were in reality registered between 1999-2004
